### PR TITLE
Improve installation portion of Vagrantfile

### PR DESCRIPTION
**Pull request by erikdw**
* Consistently use the install_package function
* Avoid installation of java 8 if already installed
* Add pseudo-code for optimizing all package_install invocations in the case of a package already being installed.
* Group together all java-related installation/setup steps
* Use the PREFIX variable consistently for output logs

### CODE REVIEW

**Comment by erikdw** [comment]
@dsKarthick & @jessicalhartog: can you please 👀 when you have a chance?

File: Vagrantfile [hunk 1]
```diff
@@ -17,39 +39,38 @@ set -e
...
+  add-apt-repository ppa:webupd8team/java
+
+  echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | sudo /usr/bin/debconf-set-selections
 
-echo "${PREFIX}Installing mesos ..."
-apt-get -o Acquire::http::Timeout=1 -o Acquire::ftp::Timeout=1 -y install mesos=0.25.0-0.2.70.ubuntu1404
-echo "Done"
+  install_package "oracle-java8-installer"
+  install_package "oracle-java8-set-default"
+  ln -sf /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so /usr/lib/libjvm.so
```
Comment (dsKarthick) [inline Vagrantfile]: just curious, why do we need this link?

**Comment by dsKarthick** [comment]
:shipit:

File: Vagrantfile (hunk 1 above)
Comment (erikdw) [inline Vagrantfile]: Just FYI, the symlink creation already existed, I'm just moving it into the "java installation" section.  However, as for "why" we need it -- seems it *might* be needed for `java`.  When I do lsof on the file it reveals java processes that reference it:

```
vagrant@master:~$ sudo lsof /usr/lib/libjvm.so
COMMAND   PID USER  FD   TYPE DEVICE SIZE/OFF   NODE NAME
java    16484 root mem    REG    8,1 16938578 920960 /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so
java    16485 root mem    REG    8,1 16938578 920960 /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so
```

**Pull request merged**